includes/data-stores/class-wcs-orders-table-subscription-data-store.php

```diff
@@ -468,17 +468,10 @@ public function update_order_meta( &$subscription ) {
 	 * @return DateTime[] The date properties which were saved to the database in array format: [ $prop_name => DateTime Object ]
 	 */
 	public function save_dates( $subscription ) {
-		global $wpdb;
-
-		$saved_dates    = [];
+		$dates_to_save  = [];
 		$changes        = $subscription->get_changes();
 		$date_meta_keys = [
-			'_schedule_trial_end',
-			'_schedule_next_payment',
-			'_schedule_cancelled',
-			'_schedule_end',
-			'_schedule_payment_retry',
-			'_schedule_start',
```

Comment:
I've removed all these date keys because we would loop over the `wcs_get_subscription_date_types()` dates anyway and add them to this array. 

That function returns   'trial_end',  'next_payment',  'last_payment', 	'cancelled',  'end' and 'start' anyway and so the only one I had to leave here was `_schedule_payment_retry`.